lib/ansible/plugins/connection/winrm.py

```diff
@@ -86,6 +87,15 @@ def __init__(self,  *args, **kwargs):
 
         super(Connection, self).__init__(*args, **kwargs)
 
+        self.host = self._play_context.remote_addr
```

Comment:
To be honest I preferred some consistency in the various connection plugins. But I really don't mind.
The difference now looks more better self-contained, so I am pleased with that as well.